addon/webextension/background/analytics.js

```diff
@@ -209,11 +234,26 @@ this.analytics = (function() {
       } else if (timingData[r.name] && match(r.end, action, label)) {
         let endTime = Date.now();
         let elapsed = endTime - timingData[r.name];
-        catcher.watchPromise(sendTiming("perf-response-time", r.name, elapsed));
+        sendTiming("perf-response-time", r.name, elapsed);
         delete timingData[r.name];
       }
     });
   }
 
+  function fetchWatcher(request) {
+    catcher.watchPromise(new Promise((resolve, reject) => {
+      request.then(response => {
+        if (response.status >= 400) {
+          let exc = new Error("Bad response from POST /timing");
+          exc.status = response.status;
+          exc.statusText = response.statusText;
+          reject(exc);
+        }
+      }).catch(error => {
+        reject(error);
+      });
```

Comment:
This could be simpler, like:

```js
function fetchWatcher(request) {
  return request.then((resp) => {
    if (!resp.ok) {
      throw new Error(`Bad response: ${resp.status} ${resp.statusText}`); // Maybe include request.url? Or just set these things on the exception like you did above
    }
    return resp;
  });
}
```

Throwing an exception in a then handler is the same as rejecting it.